samples/web/content/apprtc/gcm_register.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/python2.4
+#
+# Copyright 2015 Google Inc. All Rights Reserved.
+
+"""AppRTC GCM ID Registration
+
+This module implements user ID registration and verification.
+"""
+
+import constants
+import datetime
+import json
+import logging
+import webapp2
+
+from google.appengine.ext import ndb
+
+def get_ancestor_key():
+  return ndb.Key('GCMRecord', 'global')
+
+PARAM_USER_ID = 'userId'
+PARAM_GCM_ID = 'gcmId'
+PARAM_CODE = 'code'
+PARAM_OLD_GCM_ID = 'oldGcmId'
+PARAM_NEW_GCM_ID = 'newGcmId'
+PARAM_USER_ID_LIST = 'userIdList'
+
+class BindPage(webapp2.RequestHandler):
+  def get_message_from_json(self):
+    try:
+      message = json.loads(self.request.body)
+      return message
+    except Exception as e:
+      logging.warning('JSON load error from BindPage: error=' + str(e)
+          + ', request=' + self.request.body)
```

Comment:
Not something we need to do in this change, but we should probably log more info about the failing request, like the URI, and anything else we know about the request. We probably need a "logging/diagnostics" feature to centralize the logging and provide a uniform log format.